ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,104 @@
+# 39. Workspace Deprecation
+
+Some of the text was taken from - https://github.com/konflux-ci/architecture/pull/187
+
+## Status
+
+WiP
+
+## Context
+
+The purpose of this ADR is to revisit the workspace concept, understand the purpose
+of it, and offer an alternative implementation based on native Kubernetes APIs
+and other successful cloud native open source projects.
+
+We inherited the "workspace" concept from our early days when we expected to be integrated with the [KCP] project. Long ago (over a year at the time of this writing), we decided to remove [KCP] from the architecture of the Konflux project.
+
+In order to support multi-cluster deployment, [kubesaw] was used instead of KCP.
+
+Today, Workspaces are an important concept in the Konflux user experience. The user onboards into a workspace of their own. A team member of theirs may add them to a team workspace. As a platform administrator, if you hear about a user having trouble, you'll ask them what workspace they're working in.
```

Comment:
Why do we need to keep supporting workspaces and/or namespaces as a visible concept in Konflux? This approach means Konflux will keep having to manage its own "world" or permissions, teams  and work-areas, in parallel to elements which already full-fill this role in SCM servers.

This ends up creating all kinds of usability issues, for example a user being able to see a repo and its PRs in GitHub, but then being unable to follow the links that Konflux posts into those PRs.

I would suggest we aim towards making the UX for Konflux repo-centric like it is for SCMs rather then workspace or namespace-centric.

Reply:
That is a different idea, which actually has me intrigued. Will like to hear  more.

Reply:
The idea of a consistent permission set from source repo to release pipeline of it's build's seems great. Would konflux inherit the permissions from the upstream SCM? Or define them within konflux and it would propagate them everywhere?

Reply:
If the are no objections I'll close this thread.